.github/workflows/default.yaml

```
@@ -94,7 +94,15 @@ jobs:
     strategy:
       fail-fast: true
       matrix:
-        os: [ ubuntu-22.04, macos-14 ]
+        os:
+          - ubuntu-22.04
+          - macos-14
+        compiler:
+          - { CC: gcc, CXX: g++ }
+          - { CC: clang, CXX: clang++ }
+        exclude:
+            # on macos, we only test Xcode, aka clang
```

The above comment (pretty much retells what the code already says but) does not explain _why_ we are not testing gcc on MacOS. Please rephrase. For example, if you just did not have time or desire to investigate what gcc-specific workarounds are needed on MacOS, then we can do something like this:

```suggestion
            # TODO: Investigate what it would take to build Squid with gcc on MacOS
```

BTW, [AFAICT](https://en.wikipedia.org/wiki/Xcode), Xcode is an IDE. We are not using IDEs to test Squid builds here. AFAICT, gcc is _not_ incompatible with Xcode (even though it is no longer included with Xcode). Thus, referring to Xcode does not seem sufficient (at least).

---

AFAIK Xcode is not just an IDE. It contains the compiler, and more important all the APIs, headers, and bits needed to target the various Apple operating systems.

gcc is available on homebrew, but I don't think it's worth targeting because:
- it's highly unlikely anyone wanting to build Squid on MacOS would use gcc to build squid. It might even be that gcc relies Xcode for system includes, frameworks, and libraries
- there is little signal to be obtained in doing it - we are already testing dual-compiler platforms to avoid dialects slipping in and locking us into a specific compiler
- I tried a build with gcc-13, it failed on not finding any needed symbol in libcppunit. While I have not investigated in dhetail, a fitting explanation might be that gcc does not fully conform to the clang ABI. This would send us in library ABI version hell

---

If want this PR to rule out testing with non-clang compilers on MacOS, then we can do something like this instead of the earlier suggestion:

```suggestion
            # Non-clang testing on MacOS is too much work for very little gain.
```

---

> If you want this PR to rule out testing with non-clang compilers on MacOS, then we can do something like this instead of the earlier suggestion

This wording sounds good to me. Of course I don't rule out anything, I am just unwilling to put in the effort myself as I don't think it's worth it